### PR TITLE
Support both .yml and .yaml files

### PR DESCRIPTION
**Pull request by mpagot**
Support both .yml and .yaml files
Move get_yml_list to utils to be able to use it in pc_helper_online too.
Fix some typo and formatting in the documentation.

### CODE REVIEW

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: openqabot/utils.py [hunk 1]
```diff
@@ -21,6 +22,20 @@ def create_logger(name: str) -> logging.Logger:
     return log
 
 
+def get_yml_list(path: Path) -> List[Path]:
+    """
+    Create a list of YML filesnames from a folder
```
Comment (b10n1k) [inline openqabot/utils.py:27]: just a typo. Sorry s/filesnames/filenames/

File: openqabot/utils.py [hunk 2]
```diff
@@ -21,6 +22,20 @@ def create_logger(name: str) -> logging.Logger:
...
+    or from a single file path.
+    """
+    yml_list = []
+    extension = "*.yml"
+    if path.is_file() and path.match(extension):
```
Comment (b10n1k) [inline openqabot/utils.py:32]: Maybe it is just fine but a yaml file can have either `yml` or `yaml`. The second is not gonna ever match.

File: openqabot/utils.py [hunk 3]
```diff
@@ -21,6 +22,20 @@ def create_logger(name: str) -> logging.Logger:
...
+        yml_list.append(path)
+    else:
+        yml_list = [p for p in path.glob(extension)]
```
Comment (ricardobranco777) [inline openqabot/utils.py:35]:
```suggestion
        yml_list = list(path.glob(extension))
```

File: openqabot/utils.py (hunk 2 above)
Comment (ricardobranco777) [inline openqabot/utils.py:32]:
```suggestion
    extensions = (".yml", ".yaml")
    if path.is_file() and path.name.endswith(extensions):
```

**Review by ricardobranco777 — APPROVED** [review]
Couple of suggestions.  Otherwise LGTM.

File: openqabot/utils.py (hunk 2 above)
Comment (mpagot) [reply openqabot/utils.py:32]: seems by design https://github.com/openSUSE/qem-bot/blob/master/doc/config.md?plain=1#L4

```
All configuration files must have `.yml` extension.
```

I like the idea to extend it, eventually in a different PR.

Comment (mpagot) [reply openqabot/utils.py]: I elaborated it a little bit more. `.something` is ok for `.endswith` but it is not for `.glob` that expect a "search pattern" like `*.something` with the `*`

File: openqabot/utils.py [hunk 4]
```diff
@@ -21,6 +22,22 @@ def create_logger(name: str) -> logging.Logger:
...
+    or from a single file path.
+    """
+    yml_list = []
+    extensions = ("yml", "yaml")
+    for ext in extensions:
+        if path.is_file():
+            if path.name.endswith("." + ext):
+                yml_list.append(path)
+        else:
+            yml_list += list(path.glob("*." + ext))
```
Comment (ricardobranco777) [inline openqabot/utils.py:37]:
```suggestion
    extensions = (".yml", ".yaml")
    if path.is_file() and path.suffix in extensions:
        yml_list.append(path)
    elif path.is_dir()
        yml_list += [f for f in p.iterdir() if f.suffix in extensions]
```

Comment (mpagot) [reply openqabot/utils.py:37]: I like to keep the `if` in two separate layers like

```
 if path.is_file():
            if path.name.endswith("." + ext):
```

and not doing 
```
 if path.is_file() and path.name.endswith("." + ext):
```

for a specific use case. It is the one covered by unit test `test_get_yml_list_single_file_not_yml` : th euser provide a single file path (not a folder) but with an invalid extension. Using the `if  ... AND ... :` pattern results in it to be considered a folder.

Comment (mpagot) [reply openqabot/utils.py:37]: I like to change `else:` to `elif path.is_dir():` 

I'm not sure I like to move away from `.glob` that was the original code also before https://github.com/openSUSE/qem-bot/pull/133

**Comment by okurz** [comment]
@Mergifyio rebase

**Review by okurz — CHANGES_REQUESTED** [review]
Given all the changes I think it's time you update the PR subject and potentially also git commit messages to something more realistically describing what change this will actually bring :)

**Comment by okurz** [comment]
I see now that you have multiple changes in a single commit. If it's not too much trouble please separate the changes into individual commits

**Comment by mpagot** [comment]
> I see now that you have multiple changes in a single commit. If it's not too much trouble please separate the changes into individual commits

Done: 3 commits

**Comment by okurz** [comment]
@Mergifyio refresh

**Comment by okurz** [comment]
@openSUSE/qa-tools-team second approval needed

**Review by kalikiana — APPROVED** [review]

**Pull request merged**